### PR TITLE
Add iconEnd support to Button

### PR DESCRIPTION
**Pull request by dgalarza**
[#161512927]

image

### CODE REVIEW

File: src/Button.js [hunk 1]
```diff
@@ -7,19 +7,59 @@ import { rem, transparentize } from 'polished';
...
+        }
+
+        return `0 ${theme.space.smaller}`;
+      })(),
       height: rem('24px'),
       fontSize: theme.fontSizes.size1,
       i: {
         fontSize: rem('24px'),
-        verticalAlign: text === undefined ? 0 : verticalAlignSmall
+        verticalAlign: text === undefined ? 0 : '-5px'
```
Comment (mikecx) [inline src/Button.js]: Is there enough difference between the verticalAlignSmall at -4px and this at -5px to warrant this change?

Comment (dgalarza) [reply src/Button.js]: Yes, when looking at it again in storybook, it felt clearly misaligned.

**Comment by mikecx** [comment]
I'm seeing a slightly different view when I look at storybook.
screen shot 2018-11-01 at 12 47 23 pm

**Comment by dgalarza** [comment]
Yeah, that screenshot is a little outdated since the removal of the default margin-top on buttons causing them to bump up against one another.

**Comment by mikecx** [comment]
Can we re-add the margin to the story instead of the component?

**Pull request merged**